packages/commonwealth/client/scripts/views/pages/discussions/CommentCard/CommentCard.tsx

```diff
@@ -67,8 +72,29 @@ export const CommentCard = ({
   // actual comment
   comment,
 }: CommentCardProps) => {
+  const forceRerender = useForceRerender();
   const commentBody = deserializeDelta(editDraft || comment.text);
   const [commentDelta, setCommentDelta] = useState<DeltaStatic>(commentBody);
+  const { activeAccount: hasJoinedCommunity } = useUserActiveAccount();
+
+  const isSubscribed = () => {
+    const [newCommentSub, newReactionSub] = getCommentSubscriptions(comment);
+
+    return newCommentSub?.isActive && newReactionSub?.isActive;
+  };
+
+  const handleToggleSubscribe = async () => {
+    const [newCommentSub, newReactionSub] = getCommentSubscriptions(comment);
+
+    await handleToggleSubscription(
+      null,
+      newCommentSub,
+      newReactionSub,
+      newCommentSub?.isActive && newReactionSub?.isActive
+    );
+
+    forceRerender();
```

Comment:
Thanks for the bump @masvelio. I think we shouldn't hardcode emitters/force-renders or any side-effects into CommentCard to avoid snowballing complexity later on.

What we could do is add a `onToggleSubscription` callback and in the parent component do any emitters/force-renders or any side-effects when needed.

Reply:
[Addressed](https://github.com/hicommonwealth/commonwealth/pull/4927/commits/7d31dd206191939db63d4bb8de4aa13783216ba8).

Logic has been moved to parent component.